File: disease-detection-service/app/exceptions/app_exception.py
from app.exceptions.error_code import ErrorCode

class AppException(Exception):
    def __init__(self, error_code: ErrorCode, detail: str = None):
        super().__init__()
        self.error_code = error_code
        self.detail = detail if detail else error_code.message_key
        self.code = error_code.code
        self.status_code = error_code.status_code

File: common/src/main/java/com/leafy/common/exception/AppException.java
package com.leafy.common.exception;

import lombok.Getter;

@Getter
public class AppException extends RuntimeException {

    private final ErrorCode errorCode;

    /** Optional: the raw message from the downstream service, surfaced as-is when present. */
    private final String detail;

    public AppException(ErrorCode errorCode) {
        super(errorCode.getMessageKey());
        this.errorCode = errorCode;
        this.detail = null;
    }

    public AppException(ErrorCode errorCode, String detail) {
        super(detail != null ? detail : errorCode.getMessageKey());
        this.errorCode = errorCode;
        this.detail = detail;
    }
}